src/convertToObject.js

```diff
@@ -12,7 +12,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let array = sourceString.split(';');
+
+  const result = {};
+
+  array = array.map(item => (item.replace('"').trim()));
+
+  while (array.indexOf('') !== -1) {
+    array.splice(array.indexOf(''), 1);
+  }
+
+  array.map(item =>
+    (result[item.split(':')[0].trim()] = item.split(':')[1].trim()));
```

Comment:
Again, overcomplicated function calls. You have two same split calls.
Also, if you don't use the result of map method (new array) - it's logically better to use forEach
```suggestion
  array.forEach(item => {
    const [property, value] = item.split(':');

    result[property.trim()] = value.trim();
  });
```